automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/AuthenticatedPage.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.dsde.firecloud.page
+
+import org.openqa.selenium.WebDriver
+
+/**
+  * Base class for pages that are reachable after signing in.
+  */
+abstract class AuthenticatedPage(implicit webDriver: WebDriver) extends FireCloudView {
+
+  /**
+    * Sign out of FireCloud.
+    */
+  def signOut(): Unit = {
+    ui.clickAccountDropdown()
+    ui.clickSignOut()
+  }
+
+  def readUserEmail(): String = {
+    ui.readUserEmail()
+  }
+
+
+  trait UI {
+    private val accountDropdown = testId("account-dropdown")
+    private val accountDropdownEmail = testId("account-dropdown-email")
+    private val signOutLink = testId("sign-out")
+
+    def checkAccountDropdown: Boolean = {
+      find(accountDropdown).isDefined
+    }
+
+    def clickAccountDropdown(): Unit = {
+      click on accountDropdown
+    }
+we
+    def clickSignOut(): Unit = {
+      click on (await enabled signOutLink)
+    }
+
+    def readUserEmail(): String = {
+      await enabled accountDropdownEmail
+      find(accountDropdownEmail).get.text
+    }
+  }
+
+  /*
+   * This must be private so that subclasses can provide their own object
```

Comment:
Not sure this is true? This `ui`'s class name would be `org.broadinstitute.dsde.firecloud.page. AuthenticatedPage.ui` while any subclasses' would be `org.broadinstitute.dsde.firecloud.page.TheSubClass.ui` so it shouldn't conflict.